### PR TITLE
Collect drain statistics and report to wmstats

### PR DESCRIPTION
**Pull request by goughes**

### CODE REVIEW

**Comment by goughes** [comment]
I plan to make 4 drain status checks based on this doc: https://twiki.cern.ch/twiki/bin/viewauth/CMS/CompOpsPRAgentDoc#Drain

1. Jobs in Condor 
2. Jobs Queued Inside WMAgent
3. Workflows in the Agent (done)
4. Files To Upload

Let me know if that's what you want.

**Comment by amaltaro** [comment]
Looks good Eric.
For point 2. jobs queued inside WMAgent, the only job state allowed (in wmbs table) would be `cleanout`. Thus if all agent jobs are on that state, we could say it's ready to be redeployed (given it passes the other checks too).

For point 4. files to upload, we need to check a) files to upload to dbs, b) files to upload to phedex and c) blocks closed in phedex

**Comment by ticoann** [comment]
In the following module,
src/python/WMComponent/AgentStatusWatcher/DrainStatusPoller.py

following function can be moved to DrainStatusAPI.  I think it is easier for other script to use that function if necessary. 
```
def collectDrainInfo(self):
```

**Comment by ticoann** [comment]
For point 1. we can also check BossAir status. (bl_runjob table) to double check whether all the condor status are updated. This might not be an important check but as a testing purpose if there is conflict we can investigate.

**Comment by goughes** [comment]
wmstats

This is what wmstats looks like with drain statistics.

File: src/couchapps/WMStats/_attachments/js/minified/import-all-t1.min.js [hunk 1]
```diff
@@ -4385,7 +4402,6 @@ WMStats.namespace('ResubmissionList');
             htmlstr += "<li><b>ACDC DB Name:</b><input type='text' name='ACDCDatabase' size=30  value='wmagent_acdc'/></li>";
         }
         htmlstr += "<li><b>RequestPriority:</b> " + summary.RequestPriority + "</li>";
-        htmlstr += "<li><b>RequestType:</b> " + summary.RequestType + "</li>";
```
Comment (goughes) [inline src/couchapps/WMStats/_attachments/js/minified/import-all-t1.min.js]: Must have deleted this by accident.  I'll add it back.

**Comment by ticoann** [comment]
test this please

**Comment by amaltaro** [comment]
@goughes please let us know once you have made the final changes you mentioned during the meeting.

**Comment by goughes** [comment]
@amaltaro Ok, I'm going to put the condor stuff into a Condor API in WMCore/Services/Condor.  Then it can be expanded and used by other classes if necessary.

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
test this please

**Comment by ticoann** [comment]
test this please

**Pull request merged**